stl/inc/format

```diff
@@ -2793,6 +2909,25 @@ template <_Format_supported_charT _CharT, class _Traits>
 struct formatter<basic_string_view<_CharT, _Traits>, _CharT>
     : _Formatter_base<basic_string_view<_CharT, _Traits>, _CharT, _Basic_format_arg_type::_String_type> {};
 
+template <class _CharT, class... _Args>
+struct _Basic_format_string {
+    basic_string_view<_CharT> _Str;
+
+    template <class _Ty>
+        requires convertible_to<const _Ty&, basic_string_view<_CharT>>
+    consteval _Basic_format_string(const _Ty& _Str_val) : _Str(_Str_val) {

```

Comment:
Clang does not yet claim to support `consteval`:

```
C:\Temp>type meow.cpp
#include <stdio.h>

int main() {
#ifdef __cpp_consteval
    puts("__cpp_consteval IS defined.");
#else
    puts("__cpp_consteval is NOT defined.");
#endif
}

C:\Temp>cl /EHsc /nologo /W4 /std:c++latest meow.cpp && meow
meow.cpp
__cpp_consteval IS defined.

C:\Temp>clang-cl /EHsc /nologo /W4 /std:c++latest meow.cpp && meow
__cpp_consteval is NOT defined.
```

Thus `<source_location>` checks for support:
https://github.com/microsoft/STL/blob/3c2fd04d441d46ec9d914d9cbb621a3bac96c3a5/stl/inc/source_location#L11-L12

If downgrading to `constexpr` is possible, then we have a macro:
https://github.com/microsoft/STL/blob/3c2fd04d441d46ec9d914d9cbb621a3bac96c3a5/stl/inc/yvals_core.h#L707-L711

If Clang's limited (?) support for `consteval` is sufficient here, I guess I'm ok with unguarded usage, I just wanted to bring this up as a concern.

Reply:
the semantics change DRAMATICALLY if you change this to constexpr.

Clang claims partial support for consteval with the missing part being default parameters of consteval functions.

Reply:
Ok as-is then!

Reply:
To expand for future readers:

If the constructor is marked constexpr then if you pass a non-constant format string you get a silent fallback to runtime exceptions, this is not what the standard says to do.